### DISCUSSION TITLE
MCAP about twice slower when reading a topic from a converted ros1 file

### DISCUSSION THREAD

**Discussion by jmartinez-outrider** [foxglove/mcap#1081] [General]
I converted a rosbag file (format ros1) to mcap using the `mcap` command line.  When I read  the file it takes more than twice the time to read a given topic (in comparison to what it used to take when reading using rosbag). I'm using the python API. 

Is this something expected or am I doing something wrong?

Example of the code I'm using to read a topic.
```python
from mcap.reader import McapReader, make_reader
from mcap_ros1.decoder import DecoderFactory

with open(bag_file, "rb") as f:
    bag = make_reader(f, decoder_factories=[DecoderFactory()])
    for schema, channel, message, ros_msg in bag.iter_decoded_messages(topics=["some_topic"]):
        ....
```

**Comment by defunctzombie** [comment]
It would be helpful to reproduce if you could share the bag file that you converted, the converted mcap file, and the python code you are using to read the bag file.

**Reply by jmartinez-outrider** [reply]
I'm afraid I can't do that because of its contents (and size) but it's a bag with about 200 topics and 15 GB size. I could give it a shot at generating one with randomised data but it will take me some days to get there. I just wanted to know if this performance issue is actually an issue or it comes as a surprise.

**Comment by james-rms** [comment]
Are you able to provide `rosbag info` on this bag? i'm interested to see if it's compressed or not.

**Reply by jmartinez-outrider** [reply]
Yes I can provide info. It's not compressed. However, yesterday I did some tests with LZ4 and the compressed bag was just slightly slower (using rosbag) while mcap was significantly slower (2x and more). Is there anything else that would be useful from the info? the bag is indexed. 

I run the same 15GB going one topic at the time and reading the messages in a fairly top end machine (times are in seconds)

ROSBAG TIME: 204
ROSBAG LZ4 TIME: 360
MCAP TIME: 1486

**Reply by james-rms** [reply]
I'd be interested to see how an LZ4-compressed MCAP does - you can convert it with `mcap compress --compression lz4 in.mcap -o out.mcap`. Either way, i'll see if i can reproduce with some of our example bags on my end.
